src/Model/Behavior/LimitResultsBehavior.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Tatoeba Project, free collaborative creation of languages corpuses project
+ * Copyright (C) 2020 Tatoeba Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+namespace App\Model\Behavior;
+
+use Cake\ORM\Behavior;
+use Cake\ORM\Query;
+
+class LimitResultsBehavior extends Behavior
+{
+    /**
+     * Calculate minimal associations for a query
+     *
+     * Helper function for paginateLatest which filters the associations to
+     * load for the given query. Only the associations mentioned in the 'select'
+     * and 'order' part are necessary for calculating the lowest id we need.
+     *
+     * @param array $query The query to calculate from
+     *
+     * @return array
+     **/
+    private function getMinimalContain(Query $query) {
+        $fields = [];
+        foreach (['where', 'order'] as $clause) {
+            $clause = $query->clause($clause);
+            if ($clause) {
+                $clause->traverse(function($c) use (&$fields) {
+                    $fields[] = $c->getField();
+                });
+            }
+        }
```

Comment:
I ended up removing the check on *order* clauses because like you said we do not use it. Even if we were, I suspect it wouldn’t work anyway, because the finder assumes we order on the `id` field in descending order.